android/src/main/java/dev/openfeature/sdk/OpenFeatureAPI.kt

```diff
@@ -1,91 +1,192 @@
-        try {
-            provider.initialize(context)
-        } catch (e: Throwable) {
-            // This is not allowed to happen
+    /**
+     * Set the [FeatureProvider] for the SDK. This method will return immediately and initialize the provider in a coroutine scope
+     * When the provider is successfully initialized it will set the status to Ready.
+     * If the provider fails to initialize it will set the status to Error.
+     *
+     * This method requires you to manually wait for the status to be Ready before using the SDK for flag evaluations.
+     * This can be done by using the [statusFlow] and waiting for the first Ready status or by accessing [getStatus]
+     *
+     * @param provider the provider to set
+     * @param dispatcher the dispatcher to use for the provider initialization coroutine. Defaults to [Dispatchers.IO] if not set.
+     * @param initialContext the initial [EvaluationContext] to use for the provider initialization. Defaults to an null context if not set.
+     */
+    fun setProvider(
+        provider: FeatureProvider,
+        dispatcher: CoroutineDispatcher = Dispatchers.IO,
+        initialContext: EvaluationContext? = null
+    ) {
+        setProviderJob?.cancel()
+        this.setProviderJob = CoroutineScope(dispatcher).async {
+            setProviderInternal(provider, initialContext)
         }
     }
 
+    /**
+     * Set the [FeatureProvider] for the SDK. This method will block until the provider is initialized.
+     *
+     * @param provider the [FeatureProvider] to set
+     * @param initialContext the initial [EvaluationContext] to use for the provider initialization. Defaults to an null context if not set.
+     */
     suspend fun setProviderAndWait(
         provider: FeatureProvider,
-        dispatcher: CoroutineDispatcher,
         initialContext: EvaluationContext? = null
     ) {
-        setProvider(provider, initialContext)
-        provider.awaitReadyOrError(dispatcher)
+        setProviderInternal(provider, initialContext)
+    }
+
+    private suspend fun setProviderInternal(
+        provider: FeatureProvider,
+        initialContext: EvaluationContext? = null
+    ) {
+        this@OpenFeatureAPI.provider = provider
+        // TODO consider if stale status should emit? _statusFlow.tryEmit(OpenFeatureStatus.Stale)
+        if (initialContext != null) context = initialContext
+        try {
+            getProvider().initialize(context)
+            _statusFlow.tryEmit(OpenFeatureStatus.Ready)
+        } catch (e: OpenFeatureError) {
+            _statusFlow.tryEmit(OpenFeatureStatus.Error(e))
+        } catch (e: Throwable) {
+            _statusFlow.tryEmit(
+                OpenFeatureStatus.Error(
+                    OpenFeatureError.GeneralError(
+                        e.message ?: e.javaClass.name
+                    )
+                )
+            )
+            // TODO deal with things by setting status to Error or Fatal
```

Comment:
Is this comment still valid? I see we set the Error state in the lines just above in this same catch block